File: solutions/c/difference-of-squares/5/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int difference_of_squares(int number)
{
    int sum_of_squares = 0;
    int square_of_sum = 0;
    for (int i = 1; i <= number; i++)
    {
        sum_of_squares += i * i;
        square_of_sum += i;
    }
    square_of_sum *= square_of_sum;
    return square_of_sum - sum_of_squares;
}

File: solutions/ocaml/difference-of-squares/5/difference_of_squares.ml
let square_of_sum n = 
    (Float.of_int (n * (n + 1)/2)) ** 2.;;

let sum_of_squares n =
    (Float.of_int (n * (n + 1) * (2 * n + 1) / 6));;

let difference_of_squares n =
    square_of_sum n - sum_of_squares n ;;
